torch/testing/_internal/common_utils.py

```diff
@@ -2467,14 +2467,25 @@ def _run_with_retry(self, result=None, num_runs_left=0, report_only=True, num_re
             errors_before = 0 if result is None else len(result.errors)
             skipped_before = 0 if result is None else len(result.skipped)
 
+        def dynamo_wrapper(backend, function):
+            @functools.wraps(function)
+            def inner(*args, **kwargs):
+                # It's important to reset the Dynamo state so that previous
+                # invocations do not affect future ones.
+                torch._dynamo.reset()
+                try:
+                    return torch._dynamo.optimize(backend)(function)(*args, **kwargs)
+                finally:
+                    torch._dynamo.reset()
```

Comment:
Why after too?

Reply:
It's good to clean up after ourselves, but mechanically we don't really need it

Reply:
ok